src/main/java/com/microsoft/aad/msal4j/IBroker.java

```diff
@@ -0,0 +1,21 @@
+package com.microsoft.aad.msal4j;
+
+/**
+ * Used to define the basic set of methods that all Brokers must implement
+ */
+public interface IBroker {
+    void initializeBroker ();
+
+    Account signInSilently(PublicClientApplication clientApplication, InteractiveRequestParameters interactiveRequestParameters);
+
+    Account signInInteractively(PublicClientApplication clientApplication, InteractiveRequestParameters interactiveRequestParameters);
+
+    AuthenticationResult acquireTokenSilently(PublicClientApplication clientApplication, InteractiveRequestParameters interactiveRequestParameters, Account account);
+
+    AuthenticationResult acquireTokenInteractively(PublicClientApplication clientApplication, InteractiveRequestParameters interactiveRequestParameters, Account account);
+
+    void signOutSilently(PublicClientApplication clientApplication, InteractiveRequestParameters interactiveRequestParameters);
+
+    //TODO: too specific to MSALRuntime? Should it be split into a generic getAccount(something) and getAllAccounts(clientId)?
+    void discoverAccounts(PublicClientApplication clientApplication, InteractiveRequestParameters interactiveRequestParameters);
```

Comment:
Because all methods seem to require a `PublicClientApplication`, I wonder if this should be a constructor parameter in implementations of this interface? Do you expect for an instance of `Broker` that the `PublicClientApplication` will always be the same?

Reply:
This has been completely refactored from this initial generic design to something more like the rest of MSAL Java, where there are a series of acquireToken() methods overloaded for each type of request (leaving it up to a broker to figure out how to handle each type of request).